powersync-ja/powersync-js
# fix: reconnect powersync toggle for rn-chat demo

Fixed an issue raised by a user where reconnecting to powersync was not working.

Moved out the logic for powersync to connect/disconnect to a `useEffect` instead of a side effect in a `useMemo`.

## To test

1. Run the [react-native-supabase-group-chat](https://github.com/powersync-ja/powersync-js/tree/main/demos/react-native-supabase-group-chat).
2. Go to settings in the app and disable the sync toggle. You should see a `not connected` log message. 
3. Re-enable the sync toggle. You should now see log messages such as 
```
[PowerSyncStream] Streaming sync iteration started
[PowerSyncStream] Requesting stream from server
...
[PowerSyncStream] Stream established. Processing events
connected
```
